src/finch/tensor.py

```diff
@@ -396,7 +396,11 @@ def construct_csf(cls, arg: TupleOf3Arrays, shape: tuple[int, ...]) -> "Tensor":
         return Tensor(cls.construct_csf_jl_object(arg, shape))
 
 
-def fsprand(*args):
+def random(shape, density=0.01, random_state=None):
+    args = [*shape, density]
+    if random_state is not None:
+        rng = jl.default_rng(random_state)
+        args = [rng] + args
```

Comment:
Here we should handle the case that `random_state` is a `np.random.Generator` instance, in which case we can just generate some bytes from it to pass to `jl.default_rng`.

Reply:
Done! `jl.default_rng` only accepts an integer.

Reply:
looks good!